pkg/db/db.go

```diff
@@ -656,3 +681,51 @@ func (g *GormDB) Insert(ctx context.Context, rec *Record) error {
 		return tx.Table(g.tableName).Create(rec).Error
 	})
 }
+
+// uid is here to fulfill the value.Context interface for the transformer.
+// This is similar to authenticatedDataString from the k8s apiserver's storage interface
+// for etcd: https://github.com/kubernetes/kubernetes/blob/a42f4f61c2c46553bfe338eefe9e81818c7360b4/staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go#L63
+type uid string
+
+func (u uid) AuthenticatedData() []byte {
+	return []byte(u)
+}
+
+func (g *GormDB) encryptData(ctx context.Context, rec *Record) error {
+	gk := schema.GroupKind{Group: rec.APIGroup, Kind: rec.Kind}
+
+	if t, exists := g.transformers[gk]; exists {
+		logrus.Debugf("Encrypting data for record %s in namespace %s", rec.Name, rec.Namespace)
+		encryptedData, err := t.TransformToStorage(ctx, []byte(rec.Data.String()), uid(rec.UID))
+		if err != nil {
+			return err
+		}
+
+		rec.Data, err = json.Marshal(map[string]string{
+			"e": base64.StdEncoding.EncodeToString(encryptedData),
+		})
+		return err
+	}
+	return nil
+}
+
+func (g *GormDB) decryptData(ctx context.Context, rec *Record) error {
+	gk := schema.GroupKind{Group: rec.APIGroup, Kind: rec.Kind}
+
+	if t, exists := g.transformers[gk]; exists {
+		logrus.Debugf("Decrypting data for record %s in namespace %s", rec.Name, rec.Namespace)
+		m := &map[string]string{}
+		if err := json.Unmarshal(rec.Data, m); err != nil {
+			// If it doesn't unmarshal, then it wasn't encrypted by the transformer, so just return
+			return nil
+		}
+		data, err := base64.StdEncoding.DecodeString((*m)["e"])
```

Comment:
Nit: Can probably write it in a way without deferencing a pointer in go
 ```
 m := map[string]string{}
if err := json.Unmarshal(rec.Data, &m); err != nil {
	// If it doesn't unmarshal, then it wasn't encrypted by the transformer, so just return
	return nil
}
data, err := base64.StdEncoding.DecodeString(m["e"])
		```

Reply:
Testing this now. Should work

Reply:
It works